### PR TITLE
[2.4] Update data types in codebase

### PR DESCRIPTION
**Pull request by dgsga**

### CODE REVIEW

**Comment by dgsga** [comment]
@rdmark, should we also consider changing:
u_short to unsigned short
u_long to unsigned long
u_char to unsigned char

**Comment by rdmark** [comment]
Sure, go for it. It seems Sonarcloud complains about this?

File: include/wolfssl/wolfcrypt/settings.h [hunk 1]
```diff
@@ -1797,25 +1797,25 @@ extern void uITRON4_free(void *p) ;
     #define STRING_USER
     #define XSTRCASECMP(s1,s2) strcasecmp((s1),(s2))
     #define XSTRCMP(s1,s2) strcmp((s1),(s2))
-    #define XSTRLEN(pstr) ((CPU_SIZE_T)Str_Len((CPU_CHAR *)(pstr)))
+    #define XSTRLEN(pstr) ((CPU_SIZE_T)Str_Len((CPunsigned char *)(pstr)))
```
Comment (rdmark) [inline include/wolfssl/wolfcrypt/settings.h:1800]: What does this data type mean?

Also, this settings.h header is one that you wrote, or is it copied verbatim from wolfssl?

**Comment by rdmark** [comment]
Looks good to me. Only question is the wolfcrypt header. I'm not sure what those data types are, and if it was me I'd leave them as-is. :)

File: include/wolfssl/wolfcrypt/settings.h (hunk 1 above)
Comment (dgsga) [reply include/wolfssl/wolfcrypt/settings.h]: Well spotted as find and replace substituted U_CHAR as well as u_char! Fixed. Settings.h copied verbatim from WolfSSL

Comment (rdmark) [reply include/wolfssl/wolfcrypt/settings.h]: Search-and-replace -- that explains it!

There are a handful more instances in the same settings.h that need to be reverted, I think. Please have a look.

Comment (dgsga) [reply include/wolfssl/wolfcrypt/settings.h]: All done now!

**Review by rdmark — APPROVED** [review]

File: include/wolfssl/wolfcrypt/settings.h (hunk 1 above)
Comment (rdmark) [reply include/wolfssl/wolfcrypt/settings.h]: Cheers! Approved.

**Pull request merged**